barretenberg/cpp/pil/avm/gadgets/cmp.pil

```diff
@@ -16,6 +16,26 @@ namespace cmp(256);
     pol commit input_b;
     pol commit result;
 
+    // ========= NON FF GT Short-circuit computation ===============================
+    // If this is a non-ff check, we can short circuit (a > b) by checking
+    // 0 < a - b < 2**128 --> i.e. we just check we dont underlow which for "small" sized 128-bit number is just a 
+    // single 128 bit range check
+    // This will be constrained by the calling function - maybe through instruction decomposition
+    pol commit op_non_ff_gt;
+    // Value of a - b
+    pol commit diff;
+
+    pol IS_GT_NON_FF = op_non_ff_gt * result;
+    pol A_MINUS_B = input_a - input_b - 1;
+    pol B_MINUS_A = input_b - input_a;
+    op_non_ff_gt * (diff -  (A_MINUS_B * IS_GT_NON_FF) - (B_MINUS_A * (1 - IS_GT_NON_FF))) = 0; 
```

Comment:
As diff is only used in the "op_non_ff_gt" context, I think we can simplify the constraint by stgh like this:
`diff = input_b - input_a -1 + result * (2 * input_a - 2 * input_b +1)`